src/modules/referrals/selectors.spec.ts

```diff
@@ -0,0 +1,144 @@
+import { ReferralsState } from './reducer'
+import { getData, getLoading, getError, getInvitedUsersAccepted, getInvitedUsersAcceptedViewed, isLoadingReferrals } from './selectors'
+
+// Mock the loading selector
+jest.mock('decentraland-dapps/dist/modules/loading/selectors', () => ({
+  isLoadingType: jest.fn()
+}))
+
+describe('referrals selectors', () => {
+  let mockState: RootState
+  let mockReferralsState: ReferralsState
+  let stateWithLoading: RootState
+  let stateWithMultipleLoading: RootState
+  let stateWithError: RootState
+  let mockIsLoadingType: jest.MockedFunction<typeof isLoadingType>
+
+  beforeEach(() => {
+    mockReferralsState = {
+      data: {
+        invitedUsersAccepted: 5,
+        invitedUsersAcceptedViewed: 3
+      },
+      loading: [],
+      error: null
+    }
+
+    mockState = {
+      referrals: mockReferralsState
+    } as RootState
+
+    stateWithLoading = {
+      referrals: {
+        ...mockReferralsState,
+        loading: [fetchReferralsRequest() as AnyAction]
+      }
+    } as RootState
+
+    stateWithMultipleLoading = {
+      referrals: {
+        ...mockReferralsState,
+        loading: [fetchReferralsRequest() as AnyAction, { type: 'OTHER_ACTION' } as AnyAction]
+      }
+    } as RootState
+
+    stateWithError = {
+      referrals: {
+        ...mockReferralsState,
+        error: 'Failed to fetch referrals'
+      }
+    } as RootState
+
+    mockIsLoadingType = isLoadingType as jest.MockedFunction<typeof isLoadingType>
+  })
+
+  afterEach(() => {
+    jest.resetAllMocks()
+  })
+
+  describe("when getting the state's data", () => {
+    it('should return the referrals data', () => {
+      const result = getData(mockState)
+      expect(result).toEqual({
+        invitedUsersAccepted: 5,
+        invitedUsersAcceptedViewed: 3
+      })
+    })
+  })
+
+  describe('when getting the loading state', () => {
+    it('should return the loading state', () => {
+      const result = getLoading(mockState)
+      expect(result).toEqual([])
+    })
+
+    describe('and there are loading actions', () => {
+      it('should return the correct loading state', () => {
+        const result = getLoading(stateWithLoading)
+        expect(result).toEqual([fetchReferralsRequest()])
+      })
+    })
+
+    describe('and there are multiple loading actions', () => {
+      it('should return the correct loading state', () => {
+        const result = getLoading(stateWithMultipleLoading)
+        expect(result).toEqual([fetchReferralsRequest(), { type: 'OTHER_ACTION' }])
+      })
+    })
+  })
+
+  describe('when getting the error state', () => {
+    it('should return null when there is no error', () => {
+      const result = getError(mockState)
+      expect(result).toBeNull()
+    })
+
+    it('should return the error message when there is an error', () => {
+      const result = getError(stateWithError)
+      expect(result).toBe('Failed to fetch referrals')
+    })
```

Comment:
Would you mind re-writing the tests so they follow the standards?
There are two issues here, one is that the contexts are not defined in the `beforeEach` and the other one is that there's a `stateWithError` that is defined on the top `beforeEach`, but should be defined in the context where there's an error. I've changed it in this example to show how the state could be modified.
```suggestion
    describe('and there's no error', () => {
      beforeEach(() => {
        state.error = null
      })
    
      it('should return null', () => {
        const result = getError(mockState)
        expect(result).toBeNull()
      })
    })
    
    describe('and there's an error', () => {
      beforeEach(() => {
        state.error = null
      })
      
      it('should return the error message', () => {
        const result = getError(state)
        expect(result).toBe('Failed to fetch referrals')
      })
    })
```